filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -204,6 +229,7 @@ func lock(ctx input.Context, store *store, key string) (*resource, error) {
 		resource.Release()
 		return nil, err
 	}
+	resource.lockedVersion = resource.version
```

Comment:
You need to hold the stateMutex here. The updateOp operations are potentially run in another thread and you will have a race on that field.